modules/storages/spec/features/admin_storages_spec.rb

```diff
@@ -48,42 +48,77 @@
     expect(page).to have_select 'storages_storage[provider_type]', selected: 'Nextcloud', disabled: true
     expect(page).to have_field('storages_storage[name]', with: 'Nextcloud')
 
+    # Test the happy path for a valid server.
+    # Mock a valid response (=200) for example.com, so the host validation should succeed
     mock_server_capabilities_response("https://example.com")
     page.find('#storages_storage_name').set("NC 1")
     page.find('#storages_storage_host').set("https://example.com")
+    page.find('#storages_storage_oauth_client_id').set("0123456789ABCDEF")
+    page.find('#storages_storage_oauth_client_secret').set("123456789ABCDEF0")
     page.find('button[type=submit]').click
 
     created_storage = Storages::Storage.find_by(name: 'NC 1')
     expect(page).to have_title("Nc 1")
     expect(page.find('.title-container')).to have_text('NC 1')
     expect(page).to have_text(admin.name)
     expect(page).to have_text('https://example.com')
+    expect(page).to have_text('0123456789ABCDEF')
+    expect(page).to have_text('****')
     expect(page).to have_text(created_storage.created_at.localtime.strftime("%m/%d/%Y %I:%M %p"))
-
     page.find('.button--icon.icon-edit').click
 
     expect(page).to have_title("Edit: NC 1")
     expect(page.find('.title-container')).to have_text('Edit: NC 1')
 
-    mock_server_capabilities_response("https://other.example.com")
+    # Test the behavior of a failed validation with code 400 (Bad Request)
+    # simulating server not running Nextcloud
+    mock_server_capabilities_response("https://other.example.com", response_code: '400')
     page.find('#storages_storage_name').set("Other NC")
     page.find('#storages_storage_host').set("https://other.example.com")
+    page.find('#storages_storage_oauth_client_id').set("23456789ABCDEF01")
+    page.find('#storages_storage_oauth_client_secret').set("3456789ABCDEF012")
```

Comment:
@fraber Here, before the `set` the secret is probably `****`. So if you saved that form without changing the secret, the real secret would be lost.

Reply:
Hi, I've set a minimum length for client_id and secret to 10.
That way I make sure it's not overwritten by "****".